### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by Jeffery-Wasty**
Updated changelog to reflect changes to 12.9.0 since last changelog was published.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 ### Fixed issues
 - Changed driver behavior to allow prepared statement objects to be reused, preventing a "multiple queries are not allowed" error [#2482](https://github.com/microsoft/mssql-jdbc/pull/2482)
 - Adjusted DESTINATION_COL_METADATA_LOCK, in SQLServerBulkCopy, so that is properly released in all cases [#2484](https://github.com/microsoft/mssql-jdbc/pull/2484)
+- Fixed connection retry behavior when `connectRetryCount` is set to a value greater than 1 [#2513](https://github.com.mcas.ms/microsoft/mssql-jdbc/pull/2513)
```
Comment (lilgreenbird) [inline CHANGELOG.md:18]: fix url

Comment (lilgreenbird) [reply CHANGELOG.md:18]: this is missing some PRs see all PRs tagged for milestone https://github.com/microsoft/mssql-jdbc/pulls?q=is%3Apr+milestone%3A12.9.0+

**Review by lilgreenbird — DISMISSED** [review]
this is missing some PRs see all PRs tagged for milestone https://github.com/microsoft/mssql-jdbc/pulls?q=is%3Apr+milestone%3A12.9.0+

File: CHANGELOG.md (hunk 1 above)
Comment (Jeffery-Wasty) [reply CHANGELOG.md:18]: Odd, my URL for these PRs are different than public facing ones. Fixed.

**Review by lilgreenbird — CHANGES_REQUESTED** [review]
missed 2521

**Comment by Jeffery-Wasty** [comment]
I left that out on purpose. The JavaDoc warnings are only visible if you build the driver, and for similar changes in the past, we did not include that in the CHANGELOG.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,16 +5,19 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 
 ## [12.9.0] Preview Release
 ### Added
-- Added statement portion of configurable retry logic feature [#2396](https://github.com/microsoft/mssql-jdbc/pull/2396)
+- Added configurable retry logic feature, supporting both statement, and connection, retry [#2396](https://github.com/microsoft/mssql-jdbc/pull/2396)[#2519](https://github.com/microsoft/mssql-jdbc/pull/2519)
 - Added JDK 23 support [#2515](https://github.com/microsoft/mssql-jdbc/pull/2515)
 
 ### Changed
 - Reverted "Execute Stored Procedures Directly" feature, as well as subsequent changes related to the feature [#2488](https://github.com/microsoft/mssql-jdbc/pull/2488)
 - Changed MSAL logging from FINER to FINEST [#2489](https://github.com/microsoft/mssql-jdbc/pull/2489)
```
[the remark this answers is not in the document]
Comment (Jeffery-Wasty) [reply CHANGELOG.md:13]: My mistake.

**Review by lilgreenbird — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Pull request merged**